mobile/src/full/java/org/openhab/habdroid/core/FcmMessageListenerService.kt

```diff
@@ -123,10 +126,15 @@ class FcmMessageListenerService : FirebaseMessagingService() {
 
         if (icon != null) {
             val connection = ConnectionFactory.cloudConnection
+            val encodedIcon = Uri.encode(icon)
+            val suffix = when (getPrefs().getIconFormat()) {
+                IconFormat.Png -> "png"
+                IconFormat.Svg -> "svg"
+            }
             if (connection != null) {
                 try {
                     iconBitmap = connection.httpClient
-                            .get("images/$icon.png", timeoutMillis = 1000)
+                            .get("icon/$encodedIcon?&format=$suffix&anyFormat=true", timeoutMillis = 1000)
```

Comment:
`&format` -> `format` (`&` isn't needed right after the question mark)

Reply:
Thanks!